File: Project/src/app/service/premium.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Premium} from '../model/premium';
@Injectable({
  providedIn: 'root'
})
export class PremiumService {

 private registerUrl:string;
  constructor(private http:HttpClient) {
this.registerUrl='api/insurance/calculate_premium';
   }
public savePremium(premium:Premium){
  return this.http.post<Premium>(this.registerUrl,premium);
}
// getPremium(){
// return this.http.get<Premium[]>("http://localhost:8080/insurance/calculate_premium");
// }
}

File: Insurance Project/InsuranceProject/src/main/java/com/insurance/app/service/PremiumService.java
package com.insurance.app.service;

import com.insurance.app.model.Premium;

/** The Interface PremiumService. */
public interface PremiumService {

    /** Calculate premium.
     *
     * @param premium
     *            the premium
     * @return the int */
    int calculatePremium(Premium premium);

}
